Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1355,6 +1356,37 @@ static inline double normalizedFlexFraction(const GridTrack& track)
     return track.baseSize() / std::max<double>(1, flexFactor);
 }
 
+// FIXME: https://bugs.webkit.org/show_bug.cgi?id=282395
+void IndefiniteSizeStrategy::accumulateFlexFractionMasonry(double& flexFraction, unsigned columnIndex, GridTrackSizingDirection direction, GridLayoutState& gridLayoutState) const
+{
+    auto trackLength = m_algorithm.tracks(direction).size();
+    auto& grid = m_algorithm.grid();
+
+    for (auto trackIndex = 0u; trackIndex < trackLength; trackIndex++) {
+        GridIterator iterator(grid, direction, trackIndex);
+
+        while (auto* gridItem = iterator.nextGridItem()) {
+            auto span = m_algorithm.renderGrid()->gridSpanForGridItem(*gridItem, direction);
+            auto spanLength = span.integerSpan();
+
+            if (!GridPositionsResolver::resolveGridPositionsFromStyle(*m_algorithm.renderGrid(), *gridItem, direction).isIndefinite()) {
+                if (span.startLine() != columnIndex)
+                    continue;
+            } else {
+                auto endLine = columnIndex + spanLength;
+                if (endLine > trackLength)
+                    continue;
+
+                span = GridSpan::translatedDefiniteGridSpan(columnIndex, columnIndex + spanLength);
+            }
```

Comment:
I feel like this reads slightly better but i'll leave it up to you to however you want to structure it:


```suggestion
            bool isIndefinitePlacedItem = GridPositionsResolver::resolveGridPositionsFromStyle(*m_algorithm.renderGrid(), *gridItem, direction).isIndefinite();
            if (!isIndefinitePlacedItem && span.startLine() != columnIndex) {
                continue;

            if (isIndefinitePlacedItem) {
              if (auto endLine = columnIndex + spanLength; endLine > trackLength)
                continue;
              span = GridSpan::translatedDefiniteGridSpan(columnIndex, columnIndex + spanLength);
            }

```